### PR TITLE
update to the upgrader max version 6

### PR DESCRIPTION
**Pull request by martint-unity**
---
### Purpose of this PR
The upgrader was never updated to max version 6.
Added also a const for that.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 1]
```diff
@@ -869,7 +870,7 @@ public void OnAfterDeserialize()
 #if UNITY_EDITOR
         static void UpgradeAsset(UniversalRenderPipelineAsset asset)
         {
-            if(asset.k_AssetPreviousVersion < 5)
+            if(asset.k_AssetPreviousVersion < k_MaxAssetVersion)
```
Comment (Verasl) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:873]: Looking at this now, it seems that this needed to be still 5, as this is the version it changed from single renderer to renderer list..

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 2]
```diff
@@ -885,7 +886,7 @@ static void UpgradeAsset(UniversalRenderPipelineAsset asset)
                     asset.m_RendererData = null; // Clears the old renderer
                 }
 
-                asset.k_AssetPreviousVersion = 5;
+                asset.k_AssetPreviousVersion = k_MaxAssetVersion;
```
Comment (Verasl) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:889]: Same here, this probably means we need to re-look at this code

**Review by Verasl — CHANGES_REQUESTED** [review]
I think we changed a bit too much

File: com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs [hunk 3]
```diff
@@ -102,8 +102,9 @@ public partial class UniversalRenderPipelineAsset : RenderPipelineAsset, ISerial
         ScriptableRenderer[] m_Renderers = new ScriptableRenderer[1];
 
         // Default values set when a new UniversalRenderPipeline asset is created
-        [SerializeField] int k_AssetVersion = 5;
-        [SerializeField] int k_AssetPreviousVersion = 5;
+        const int k_MaxAssetVersion = 6;
+        [SerializeField] int k_AssetVersion = k_MaxAssetVersion;
+        [SerializeField] int k_AssetPreviousVersion = k_MaxAssetVersion;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:107]: Shouldn't this be `k_MaxAssetVersion - 1`?

Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:107]: This logic is rather confusing. Is k_MaxAssetVersion current pipeline asset version?

Comment (martint-unity) [reply com.unity.render-pipelines.universal/Runtime/Data/UniversalRenderPipelineAsset.cs:107]: It is very confusing indeed.

**Review by Verasl — APPROVED** [review]
LGTM, should solve some issues without having to remake the entire 'upgrader' for now

**Comment by martint-unity** [comment]
closing due not feeling safe to land.

**Pull request closed without merging**